packages/yarnpkg-fslib/sources/ZipFS.ts

```diff
@@ -313,6 +315,10 @@ export class ZipFS extends BasePortableFakeFS {
     if (rc === -1)
       throw this.makeLibzipError(this.libzip.getError(this.zip));
 
+    // zip_close doesn't persist empty archives
+    if (!this.baseFs.existsSync(this.path))
+      this.baseFs.writeFileSync(this.path, makeEmptyArchive());
+
```

Comment:
Could we instead check if `this.entries` is empty and just write the empty archive directly?

Reply:
Done, good idea :+1: